src/main/java/codeu/controller/ChatServlet.java

```diff
@@ -161,49 +161,51 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 		}
 
 		Part messagePart = request.getPart("message");
-		Scanner s = new Scanner(messagePart.getInputStream());
+		if (messagePart != null) {
+			Scanner s = new Scanner(messagePart.getInputStream());
 
-		if (s.hasNextLine()) {
-			String messageContent = s.nextLine();
+			if (s.hasNextLine()) {
+				String messageContent = s.nextLine();
 
-			// allows users to enter basic HTML tags that are not a threat to security
-			String HTMLMessageContent = clean(messageContent);
+				// allows users to enter basic HTML tags that are not a threat to security
+				String HTMLMessageContent = clean(messageContent);
 
-			Message message = new Message(UUID.randomUUID(), conversation.getId(), user.getId(), HTMLMessageContent,
-					Instant.now());
+				Message message = new Message(UUID.randomUUID(), conversation.getId(), user.getId(), HTMLMessageContent,
+						Instant.now());
 
-			messageStore.addMessage(message);
+				messageStore.addMessage(message);
 
+			}
+			s.close();
 		}
-		s.close();
-
 		// Add image message if there is one
 
 		Part filePart = request.getPart("upload");
-		InputStream fileStream = filePart.getInputStream();
-
-		// fileStream.available() returns the number of bytes that are ready to read,
-		// so if it is 0, then no file was uploaded. We also cap this at 1MB because
-		// that is the max data size that can be saved as a property in the datastore.
-		int bytesAvailable = fileStream.available();
-		if (bytesAvailable > 0 && bytesAvailable < 1000000) {
```

Comment:
Will it return an error if the file is greater than 1MB? It might be nice for the user to know that their file is too big to be uploaded.